### DISCUSSION TITLE
`flux diff` with remote cluster [`.spec.kubeConfig`]

### DISCUSSION THREAD

**Discussion by milas** [fluxcd/flux2#5065] [Q&A] [answered]
Currently, if I have a `Kustomization` that targets another cluster using `.spec.kubeConfig`, when running `flux diff`, it seems to ignore that and pretend it's being applied to the _current_ cluster.

I realize full client-side multi-cluster diff is likely impractical, but does anyone have any tips here? Using `--ignore-paths` is one option, though it causes the diff to state that those resources are being _deleted_, which is a bit scary/confusing. [I've been using `flux diff` to post a comment to PRs, which has been helpful for devs as a quick sanity check/preview ahead-of-time.]

IMO, Ideally, such resources could be skipped & perhaps show something like:
```text
[!] ks-foo uses Secret/my-kubeconfig, diff not supported
```

Happy to try to work on a PR for something along those lines if it seems feasible?

But I'm also interested in any creative solutions/workarounds from others, and of course I'd love to find out I'm holding it wrong and Flux already supports this.

**Comment by stefanprodan** [comment, marked as answer]
I agree that we should error out, maybe the message could be `diff not supported for remote clusters`

**Reply by milas** [reply]
Thanks - opened a PR @ https://github.com/fluxcd/flux2/pull/5071 , can continue impl discussion there

**Comment by maboehm** [comment]
I was also looking into diffing `Kustomizations` that are applied to a different cluster, and came up with this proposal:
https://github.com/fluxcd/flux2/compare/main...maboehm:flux2:diff-kubeconfig
(I did not consider the recursive case just yet)

In my approach, the CLI tries to use the secret from `spec.kubeConfig` when diffing, but also offers a `--target-kubeconfig` for cases where the in-cluster kubeconfig only works from inside the cluster.

Would that be something worth adding? It's still not perfect, but still an improvement IMO.

**Reply by Preisschild** [reply]
Do you plan on opening a PR with that? IMO it would be extremely helpful for remote-cluster use cases
